flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -376,6 +376,10 @@ public void cancel() {
 		return executionGraph.getTerminationFuture().thenApply(FunctionUtils.nullFn());
 	}
 
+	@Override
+	public void handleGlobalFailure(final Throwable cause) {
```

Comment:
Because this method should not be considered optional, I would declare this method abstract and make the legacy scheduler throw `IllegalStateException`

Reply:
Ok. Done.